### ISSUE TITLE
To elaborate on [my point above](https://github.com/nextcloud/spreed/pull/10105#pullrequestreview-1564772478):

### ISSUE THREAD

**Issue by marcoambrosini** [nextcloud/spreed#10202]
To elaborate on [my point above](https://github.com/nextcloud/spreed/pull/10105#pullrequestreview-1564772478):

This is a pattern that we use in several places in Nextcloud that I think we should break: **creating baby forms within action menus**. We shouldn't include forms, especially if they involve multi-selects and pickers, into tiny action menus. 

Having a text input with an inline submit button is **probably** ok but far from ideal, but in this case we have a baby form with a submit button that looks exactly like an actionbutton, creating visual confusion and usability issues. For example here  there's no way to guarantee that the picker itself won't block the submit button below. (see screenshot)

This came up in a recent design call and there was some consensus on that we should probably even deprecate these complex action input elements. cc @nextcloud/designers 

<img width="569" alt="Screenshot 2023-08-09 at 17 38 06" src="https://github.com/nextcloud/spreed/assets/26852655/d87088da-96cd-4557-bd78-dea94247cb17">

_Originally posted by @marcoambrosini in https://github.com/nextcloud/spreed/issues/10105#issuecomment-1670928740_

**Comment by marcoambrosini** [comment]
I agree we should be consistent, but we should break this pattern and adopt a better one. This is another instance in mail:

<img width="475" alt="Screenshot 2023-08-14 at 17 50 24" src="https://github.com/nextcloud/spreed/assets/26852655/19a7a193-f177-43dc-a684-fb34d61a27e1">

--

To provide a counter example. [In the files reminders issue, Gmail was used as inspiration](https://github.com/nextcloud/server/issues/39181#issue-1791008111), but there clicking "pick date and time" in the action menu brings up a nice and tidy dialog where everything is clearly laid out.

<img width="515" alt="Screenshot 2023-08-14 at 17 52 05" src="https://github.com/nextcloud/spreed/assets/26852655/f62698d3-659a-40a6-b895-d608aa5eda21">

**Comment by jancborchardt** [comment]
This seems to be an issue for the components library though, not only for Talk?

**Comment by marcoambrosini** [comment]
The issue in the component library is that we have an actioninput type date.

**Comment by skjnldsv** [comment]
Raising a question here.
With the amount of work we've put through going for the native datetime picker, we wanna step away from it again and use another library to render it in a modal?

**Comment by nickvergessen** [comment]
No, just not use kt in NcActions directly, but basically it should open a modal and there we use the native datetime picker iirc

**Comment by skjnldsv** [comment]
Exactly what I'm saying, you cannot render the native picker as is.
The picker is always bound to an `<input type="date">`.

**Comment by nickvergessen** [comment]
No, one level more 😅

NcModal will contain the input field of type date and the save button, instead of 2 Action components interacting with each other 🙃

**Comment by skjnldsv** [comment]
> NcModal will contain the input field of type date and the save button, instead of 2 Action components interacting with each other 🙃

I don't think this is the request.

See latest Marco [mockup](https://github.com/nextcloud/spreed/assets/14975046/9d8be7e6-9df0-4d54-83d6-feeaff041ba2), I think the idea was to have the layout rendered inline in the modal

**Comment by nickvergessen** [comment]
I think that is the native look on MacOS 😅

**Comment by skjnldsv** [comment]
Please clarify @marcoambrosini @nextcloud/designers

**Comment by marcoambrosini** [comment]
@skjnldsv I mean putting the native date picker in a modal with confirmation button, so yes what Joas said above

**Comment by skjnldsv** [comment]
Implementation in https://github.com/nextcloud/server/pull/41010

![image](https://github.com/nextcloud/spreed/assets/14975046/be53925c-0255-4a88-8d07-a3cd68d0399a)

**Comment by marcoambrosini** [comment]
Beautiful @skjnldsv :)
